src/external/monaco.flux.server.ts

```diff
@@ -44,142 +37,52 @@ import {
   SymbolInformation,
   TextEdit,
 } from 'monaco-languageclient/lib/services'
-import {Server} from '@influxdata/flux-lsp-browser'
-
-import {BUCKET_LIMIT} from 'src/resources/constants'
-
-type BucketCallback = () => Promise<string[]>
-type MeasurementsCallback = (bucket: string) => Promise<string[]>
-type TagKeysCallback = (bucket: string) => Promise<string[]>
-
-export interface WASMServer extends Server {
-  register_buckets_callback: (BucketCallback) => void
-  register_measurements_callback: (MeasurementsCallback) => void
-  register_tag_keys_callback: (TagKeysCallback) => void
-}
-
-import {format_from_js_file} from '@influxdata/flux'
-
-// NOTE: parses table then select measurements from the _value column
-const parseQueryResponse = response => {
-  const {table} = fromFlux(response.csv)
-  return table.getColumn('_value', 'string') || []
-}
-
-const queryMeasurements = async (orgID, bucket) => {
-  if (!orgID || orgID === '') {
-    throw new Error('no org is provided')
-  }
-
-  const query = `import "influxdata/influxdb/v1"
-      v1.measurements(bucket:"${bucket}")`
-
-  event('runQuery', {context: 'monaco'})
-  const raw = await runQuery(orgID, query).promise
-  if (raw.type !== 'SUCCESS') {
-    throw new Error('failed to get measurements')
-  }
-
-  return raw
-}
-
-const queryTagKeys = async (orgID, bucket) => {
-  if (!orgID || orgID === '') {
-    throw new Error('no org is provided')
-  }
-
-  const query = `import "influxdata/influxdb/v1"
-      v1.tagKeys(bucket:"${bucket}")`
-
-  event('runQuery', {context: 'monaco'})
-  const raw = await runQuery(orgID, query).promise
-  if (raw.type !== 'SUCCESS') {
-    throw new Error('failed to get tagKeys')
-  }
-
-  return raw
-}
-
-const queryTagValues = async (orgID, bucket, tag) => {
-  if (!orgID || orgID === '') {
-    throw new Error('no org is provided')
-  }
-
-  const query = `import "influxdata/influxdb/v1"
-      v1.tagValues(bucket:"${bucket}", tag: "${tag}")`
-
-  event('runQuery', {context: 'monaco'})
-  const raw = await runQuery(orgID, query).promise
-  if (raw.type !== 'SUCCESS') {
-    throw new Error('failed to get tagKeys')
-  }
-
-  return raw
-}
+import {Lsp} from '@influxdata/flux-lsp-browser'
+import {ProtocolToMonacoConverter} from 'monaco-languageclient/lib/monaco-converter'
 
+import {format_from_js_file} from '@influxdata/flux-lsp-browser'
 export class LSPServer {
-  private server: WASMServer
+  private server: Lsp
   private messageID: number = 0
   private documentVersions: {[key: string]: number} = {}
   public store: Store<AppState & LocalStorage>
+  private p2m: ProtocolToMonacoConverter
 
-  constructor(server: WASMServer, reduxStore = getStore()) {
+  constructor(server: Lsp, reduxStore = getStore()) {
     this.server = server
-    this.server.register_buckets_callback(this.getBuckets)
```

Comment:
these handlers are not available on the new library yet. They will be re-introduced as notification messages which we will handle in the `onMessage` handler

Reply:
why would the autocomplete functionality be replaced by notification messages?